test/framework/deployment_helpers.go

```diff
@@ -552,9 +528,36 @@ func DeployUnevictablePod(ctx context.Context, input DeployUnevictablePodInput)
 
 	AddPodDisruptionBudget(ctx, AddPodDisruptionBudgetInput{
 		Namespace: input.Namespace,
-		ClientSet: workloadClient,
+		ClientSet: input.WorkloadClusterProxy.GetClientSet(),
 		Budget:    budget,
 	})
+}
+
+// DeployPodAndWait will deploy a Deployment on a ControlPlane or MachineDeployment.
+func DeployPodAndWait(ctx context.Context, input DeployPodAndWaitInput) {
+	Expect(input.DeploymentName).ToNot(BeNil(), "Need a deployment name in DeployUnevictablePod")
```

Comment:
```suggestion
	Expect(input.DeploymentName).ToNot(BeNil(), "Need a deployment name in DeployPodAndWait")
```
nit, same in the few lines below